payment/datastore.go

```diff
@@ -143,6 +146,75 @@ func (pg *Postgres) GetOrder(orderID uuid.UUID) (*Order, error) {
 	return &order, nil
 }
 
+// GetPagedMerchantTransactions - get a paginated list of transactions for a merchant
+func (pg *Postgres) GetPagedMerchantTransactions(
+	ctx context.Context, merchantID uuid.UUID, pagination *inputs.Pagination) (*[]Transaction, int, error) {
+	var (
+		total int
+		err   error
+	)
+
+	countStatement := `
+			SELECT count(t.*) as total
+			FROM transactions as t
+				INNER JOIN orders as o ON o.id = t.order_id
+			WHERE o.merchant_id = $1`
+
+	// get the total count
+	row := pg.DB.QueryRow(countStatement, merchantID)
+
+	if err := row.Scan(&total); err != nil {
+		return nil, 0, err
+	}
+
+	getStatement := `
+		SELECT t.*
+		FROM transactions as t
+			INNER JOIN orders as o ON o.id = t.order_id
+		WHERE o.merchant_id = $1
+		`
+
+	// $ numbered params for query
+	params := []interface{}{
+		merchantID,
+	}
+
+	orderBy := pagination.GetOrderBy()
+	if orderBy != "" {
+		getStatement += fmt.Sprintf(" ORDER BY %s", orderBy)
```

Comment:
Is there any sanitization for these attributes? I think we filter them in the controller which is good enough?

Reply:
line 79 is where the order parameters are sanitized, based on what is passed in to inputs.NewPaginiation variadic parameter.  Unfortunately with the order by statement, i don't think you can pass those in through the go libary?

Reply:
No, don't think so either. I'm good with this 👍